sdk/core/core-http/lib/credentials/rawTokenCredential.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { TokenCredential, GetTokenOptions, AccessToken } from "@azure/core-auth";
+
+/**
+ * A TokenCredential that always returns the given token.
+ */
+export class RawTokenCredential implements TokenCredential {
```

Comment:
Is this like a helper class to create something that implements the required interface, used by users who have their own way of getting the token which doesnt have a mechanism to get refreshed?

Reply:
Yep, that's right.   I'll add extra clarification into the class docstring, something like:

> This class can be used when the access token is already known or can be retrieved from an outside source.

Reply:
@amarzavery Do we really have a user scenario for this?

Reply:
@daviwil To your below question, my answer would be yes.

> Should RawTokenCredential live in @azure/core-auth so that it can be used outside of @azure/core-http, like in @azure/core-amqp?

Mainly because `TokenCredential` lives in `core-auth`, so it makes sense for a simple class that implements the interface lives right there as well

Reply:
Moved this to `core-auth` in the latest changes.